### PR TITLE
Update changelog for 0.10.0 release

### PR DESCRIPTION
**Pull request by consideRatio**
Lets make a release finally!

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,32 +4,11 @@ Here you can find upgrade changes in between releases and upgrade instructions.
...
-  for more details.
-- Deploying the Helm chart in a cluster with a PodSecurityPolicy active is now
-  easier, because the pods' containers now have `securityContext` set on them to
-  run with relatively low permissions which are also configurable if needed.
-- The `autohttps` pod that is running to acquire TLS certificates if
-  `proxy.https.type=letsencrypt` is now more reliably acquiring certificates. If
-  you currently have such issue, do `kubectl delete deploy/autohttps` and
-  `kubectl delete secret proxy-public-tls-acme` and then deploy the Helm chart
-  again with `helm upgrade`.
+This release makes the deployment more robust, and enhance users ability to
```
Comment (choldgraf) [inline CHANGELOG.md:9]:
```suggestion
This release makes the deployment more robust, and enhances users ability to
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -74,7 +53,48 @@ can help deploy the Helm chart in a cluster with security requirements.
   `hub-secret`, which will affect anyone who use the `hub.existingSecret`
   option.
 
+#### Release highlights
+
+- An ability to configure environment variables in pods with a k8s native syntax
```
Comment (choldgraf) [inline CHANGELOG.md:58]:
```suggestion
- **Environment variables in pods with K8S config**. An ability to configure environment variables in pods with a k8s native syntax
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -74,7 +53,48 @@ can help deploy the Helm chart in a cluster with security requirements.
...
+  has been added. This allows you to reference and mount a field in a k8s Secret
+  as an environment variable for example. For more information, read [about
+  extraEnv](https://zero-to-jupyterhub.readthedocs.io/en/latest/resources/reference.html#singleuser-extraenv)
+  in the configuration reference.
+- imagePullSecrets for all the pods in the Helm chart can now be configured
```
Comment (choldgraf) [inline CHANGELOG.md:63]:
```suggestion
- **Configure secrets for all pods via the helm chart**. imagePullSecrets for all the pods in the Helm chart can now be configured
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -74,7 +53,48 @@ can help deploy the Helm chart in a cluster with security requirements.
...
+  chart wide. See the configuration reference about
+  [imagePullSecret](https://zero-to-jupyterhub.readthedocs.io/en/latest/resources/reference.html#imagepullsecret)
+  and
+  [imagePullSecrets](https://zero-to-jupyterhub.readthedocs.io/en/latest/resources/reference.html#imagepullsecrets)
+  for more details.
+- Deploying the Helm chart in a cluster with a PodSecurityPolicy active is now
```
Comment (choldgraf) [inline CHANGELOG.md:69]:
```suggestion
- **Pod security is easier to use and configure**. Deploying the Helm chart in a cluster with a PodSecurityPolicy active is now
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -74,7 +53,48 @@ can help deploy the Helm chart in a cluster with security requirements.
...
+  easier, because the pods' containers now have `securityContext` set on them to
+  run with relatively low permissions which are also configurable if needed.
+- The `autohttps` pod that is running to acquire TLS certificates if
```
Comment (choldgraf) [inline CHANGELOG.md:72]:
```suggestion
- **More reliable TLS certificates**. The `autohttps` pod that is running to acquire TLS certificates if
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -74,7 +53,48 @@ can help deploy the Helm chart in a cluster with security requirements.
...
+  `proxy.https.type=letsencrypt` is now more reliably acquiring certificates. If
+  you currently have such issue, do `kubectl delete deploy/autohttps` and
+  `kubectl delete secret proxy-public-tls-acme` and then deploy the Helm chart
+  again with `helm upgrade`.
+
+#### Notable dependencies updated
```
Comment (choldgraf) [inline CHANGELOG.md:78]: this is great - I bet we can auto-generate this in the future (though I suppose we can also just copy/paste this table in the future!)

**Review by choldgraf — COMMENTED** [review]
This looks great - just a few quick thoughts about the highlights section and I think it's GTG

File: CHANGELOG.md (hunk 6 above)
Comment (consideRatio) [reply CHANGELOG.md:78]: I think so too, but as long as we have a template to copy/paste, most of the work is done. I'm more afraid of the maintaining the automated infrastructure than updating this manually between releases :p

**Pull request merged**